src/compute_offsets.cpp

```diff
@@ -121,6 +121,13 @@ void compute_offsets(TypedAST::AccessExpression* ast, int frame_offset) {
 	compute_offsets(ast->m_record, frame_offset);
 }
 
+void compute_offsets(TypedAST::MatchExpression* ast, int frame_offset) {
+	compute_offsets(&ast->m_matchee, frame_offset);
+
+	for (auto& expr : ast->m_expressions)
+		compute_offsets(expr.second, frame_offset);
+}
```

Comment:
Since piggybacking on function literals is nasty, we should do something different here.

Something like starting a new block for every match-case, and 'pushing' space for the name that's used on every case.

Reply:
You can think about what should happen by pretending to inline `compute_offsets` of function_literals and call_expressions into this function, and see what things become redundant.

example: we don't need captures because the call happens surrounded by the same scope as the literal creation.

Reply:
The same applies to `match_identifiers`